charmhelpers/fetch/ubuntu.py

```diff
@@ -457,7 +545,55 @@ def _add_apt_repository(spec):
     if '{series}' in spec:
         series = lsb_release()['DISTRIB_CODENAME']
         spec = spec.replace('{series}', series)
-    _run_with_retries(['add-apt-repository', '--yes', spec])
+    # software-properties package for bionic properly reacts to proxy settings
+    # passed as environment variables (See lp:1433761). This is not the case
+    # LTS and non-LTS releases below bionic.
+    _run_with_retries(['add-apt-repository', '--yes', spec],
+                      cmd_env=_env_proxy_settings(['https']))
+
+
+def _env_proxy_settings(selected_settings=None):
+    """
```

Comment:
Please change docstring comment to title + description.  Also title on same line as `"""`.

Reply:
@ajkavanagh better now?